src/subscript/sunsch/sunsch.py

```diff
@@ -102,139 +71,26 @@ def _shuffle_start_refdate(config: dict) -> dict:
-                        },
-                        "template": {
-                            MK.Description: (
-                                "Template file in which substitution will "
-                                "take place before it is inserted"
-                            ),
-                            MK.Type: types.String,
-                            MK.AllowNone: True,
-                            MK.ElementValidators: (_is_existing_file,),
-                        },
-                        "days": {
-                            MK.Description: (
-                                "Days after refdate/startdate at which "
-                                "insertion should take place"
-                            ),
-                            MK.Type: types.Number,
-                            MK.AllowNone: True,
-                        },
-                        "string": {
-                            MK.Description: ("A string to insert, instead of filename"),
-                            MK.Type: types.String,
-                            MK.AllowNone: True,
-                        },
-                        "substitute": {
-                            MK.Description: (
-                                "Key-value pairs for substitution in a template"
-                            ),
-                            MK.Type: types.Dict,
-                            MK.Content: {
-                                MK.Key: {
-                                    MK.Description: "Template key name",
-                                    MK.AllowNone: False,
-                                    MK.Type: types.String,
-                                },
-                                MK.Value: {
-                                    MK.AllowNone: "Value to insert in template",
-                                    MK.AllowNone: False,
-                                    # Since we allow both numbers and strings here,
-                                    # it is converted to a string as configsuite
-                                    # only allows one type.
-                                    MK.Transformation: _to_string,
-                                    MK.Type: types.String,
-                                },
-                            },
-                        },
-                    },
-                }
-            },
-        },
-    },
-}
-
-
-def get_schema() -> dict:
-    """Return the ConfigSuite schema"""
-    return CONFIG_SCHEMA_V2
+class InsertStatement(BaseModel):
+    date: Optional[datetime.date] = None
+    filename: Optional[FilePath] = None
+    template: Optional[FilePath] = None
+    days: Optional[float] = None
+    string: Optional[str] = None
+    substitute: Optional[Dict[str, Union[str, float, str]]] = None
+
+
+class SunschConfig(BaseModel):
+    files: Optional[List[FilePath]] = None
+    output: Optional[str] = "-"
+    startdate: Union[datetime.date, datetime.datetime]
```

Comment:
Would add: _ensure_start_refdate as a model validator, something like:
```python
    @model_validator(mode="after")
    def ensure_start_date(self) -> Self:
        if not self.startdate:
            if self.starttime:
                self.startdate = self.starttime
            elif self.refdate:
                self.startdate = self.refdate
            else:
                self.startdate = datetime.date(1900, 1, 1)
        if not self.starttime:
            self.starttime = self.startdate
       if not self.refdate:
            self.refdate = self.startdate
        return self
```
could potentially also be a pre-validation.

Reply:
Done, as a pre-validation. After-validation was not possible as it will invalidate based on the values given before this after_validator is run.

Reply:
Tried also as after-validation by setting the date fields to be Optional, allowing such an after-validator:
```
    @model_validator(mode="after")
    def set_date_defaults(self) -> "SunschConfig":
        self.startdate = (
            self.startdate
            or self.starttime
            or self.refdate
            or datetime.date(1900, 1, 1)
        )
        self.starttime = self.starttime or datetime_from_date(self.startdate)
        self.refdate = self.refdate or self.startdate
        return self
```

This was even prettier but gave uglier changes to the rest of the code as all parts have to handle the Optional fields to make mypy happy. Will stick to the before-validator.

Reply:
I think in general before-validators are more practical for this kind of argument inference stuff